### PR TITLE
Add support for forcing crafting hometown 

### PR DESCRIPTION
**Pull request by vtcifer**
This adds support for several new yaml settings to allow hometown overrides on where to do crafting, similar to how it is handled with force_healer_town and saferoom.

I've tested for shaping and remedies, and works perfect, but could probably stand to be tested with others as well.

### CODE REVIEW

File: remedy.lic [hunk 1]
```diff
@@ -17,11 +17,14 @@ class Remedy
 
   def initialize
     @settings = get_settings
+    @hometown = @settings.hometown
```
Comment (rpherbig) [inline remedy.lic:20]: What would you think about something like `@hometown = @settings.force_remedies_crafting_town || @settings.force_crafting_town || @settings.hometown`? Do you think that makes the order of precedence clearer?

Comment (vtcifer) [reply remedy.lic:20]: I liked the symmetry with how the other scripts work (and in some cases HAS to work due to how smith works) to keep the code logic as close to the same, but am not opposed to switching it up if it's more in line with ruby style standards.

Comment (rpherbig) [reply remedy.lic:20]: It's not a big deal

**Comment by vtcifer** [comment]
After review, some additional scripts (helper scripts mostly) need to be updated.  Will try to get some work done on this in the next few weeks, but pulling back to draft for now.

**Comment by vtcifer** [comment]
Stale forked repo - will redo at a later time

**Pull request closed without merging**